ClLike/cl_like/bacco.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+import pyccl as ccl
+import baccoemu
+
+
+class BaccoCalculator(object):
+    """ This class implements a set of methods that can be
+    used to compute the various components needed to estimate
+    perturbation theory correlations. These calculations are
+    currently based on FAST-PT
+    (https://github.com/JoeMcEwen/FAST-PT).
+
+    Args:
+        a_arr (array_like): array of scale factors at which
+            growth/bias will be evaluated.
+    """
+    def __init__(self, log10k_min=np.log10(0.008), log10k_max=np.log10(0.5), nk_per_decade=20,
+                 a_arr=None):
+        nk_total = int((log10k_max - log10k_min) * nk_per_decade)
+        self.ks = np.logspace(log10k_min, log10k_max, nk_total)
+        if a_arr is None:
+            a_arr = 1./(1+np.linspace(0., 4., 30)[::-1])
+        self.a_s = a_arr
+        self.lbias = baccoemu.Lbias_expansion(allow_extrapolate=False)
+
+    def update_pk(self, cosmo):
+        """ Update the internal PT arrays.
+
+        Args:
+            pk (array_like): linear power spectrum sampled at the
+                internal `k` values used by this calculator.
+        """
+        h = cosmo['h']
+        cospar = {
+            'omega_cold': cosmo['Omega_c'] + cosmo['Omega_b'],
+            'omega_baryon': cosmo['Omega_b'],
+            'sigma8_cold': cosmo.sigma8(),
+            'ns': cosmo['n_s'],
+            'hubble': h,
+            'neutrino_mass': np.sum(cosmo['m_nu']),
+            'w0': cosmo['w0'],
+            'wa': cosmo['wa']}
+        k_for_bacco = self.ks/h
+        self.mask_ks_for_bacco = np.squeeze(np.where(k_for_bacco <= 0.75))
```

Comment:
Couldn't this be problematic? Should it raise an error if the user needs to go beyond 0.75? (or at least need to set a flag to force it to go beyond the valid scales?)